pkg/activator/handler/transport.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/http2"
+	"knative.dev/control-protocol/pkg/certificates"
+	netcfg "knative.dev/networking/pkg/config"
+	pkgnet "knative.dev/pkg/network"
+	activatorconfig "knative.dev/serving/pkg/activator/config"
+)
+
+func verifySanConnection(san string) func(tls.ConnectionState) error {
+	return func(cs tls.ConnectionState) error {
+		if cs.PeerCertificates == nil {
+
+			return errors.New("server certificate could not be verified during VerifyConnection. No PeerCertificates provided")
+		}
+		for _, name := range cs.PeerCertificates[0].DNSNames {
+			if name == san {
+				return nil
+			}
+		}
+		return fmt.Errorf("server with SAN %s does not have a matching name in certificate.  Names provided in certificate: %s", san, cs.PeerCertificates[0].DNSNames)
+	}
+}
+
+// tlsWrapper is a tls.Config wrapper with a TLS dialer for HTTP1
+type tlsWrapper struct {
+	tlsConf *tls.Config
+}
+
+// dialTLSContextHTTP1 handles HTTPS:HTTP1 dialer
+func (tw *tlsWrapper) dialTLSContextHTTP1(ctx context.Context, network, addr string) (net.Conn, error) {
+	return dialTLSContext(ctx, network, addr, tw.tlsConf)
+}
+
+// dialTLSContext handles HTTPS:HTTP1 and HTTP2 dialers
+// Depends on the activator handler's RevIDFrom
+func dialTLSContext(ctx context.Context, network, addr string, tlsConf *tls.Config) (net.Conn, error) {
+	config := activatorconfig.FromContext(ctx)
+	trust := config.Trust
+	if trust != netcfg.TrustDisabled {
+		tlsConf = tlsConf.Clone()
+		san := certificates.LegacyFakeDnsName
+		if trust != netcfg.TrustMinimal {
+			revID := RevIDFrom(ctx)
+			san = certificates.DataPlaneRoutingName(revID.Namespace)
```

Comment:
What happens when we the activator hits the the cluster local gateway for internal L7 routing?

Reply:
Please elaborate. 
If the question is what happens when the request comes via the local ingress? If so, it is treated the same as any other ingress at the activator. But I suspect I am not following what you are asking.

Reply:
Actually nm - this isn't a scenario that happens